### PR TITLE
Improve testing guidance

### LINKED ISSUE #244: Error loading AssetManifest.json in tests for latest package version 3.0.0
## Steps to Reproduce

Execute `flutter test --no-test-assets fonts_test.dart` on the code sample 

**Expected results:** 

Test should pass

**Actual results:** 

Test failed with the following error:

```
Error loading AssetManifest.json. Please file an issue at https://github.com/material-foundation/google-fonts-flutter/issues/new/choose.
Error: google_fonts was unable to load font ABeeZee-Regular because the following exception occured:
Unable to load asset: AssetManifest.json
See https://docs.flutter.dev/development/data-and-backend/networking#platform-notes.
00:01 +0 -1: loading /<path_to_test>/fonts_test.dart [E]                                                                                                                         
  Failed to load "/<path_to_test>/fonts_test.dart": Unable to load asset: AssetManifest.json
  package:flutter/src/services/asset_bundle.dart 237:7    PlatformAssetBundle.load
  ===== asynchronous gap ===========================
  package:flutter/src/services/asset_bundle.dart 72:27    AssetBundle.loadString
  ===== asynchronous gap ===========================
  package:google_fonts/src/asset_manifest.dart 31:11      AssetManifest._loadAssetManifestJson
  ===== asynchronous gap ===========================
  package:google_fonts/src/google_fonts_base.dart 137:31  loadFontIfNecessary
  
00:01 +0 -1: test fonts - did not complete [E]                                                                                                                                     
00:01 +0 -1: Some tests failed. 
```

As a matter of fact, v2 would also output an error message, but the test doesn't fail. Below is the output for running the same test in `2.3.3`:

```
Error loading AssetManifest.json, e: Unable to load asset: AssetManifest.json
00:01 +1: All tests passed!
```

I also thought that providing the flag `--no-test-assets` should avoid running into errors like this. Let me know if there's another flag that I should be using when running the tests.

<details>
<summary>Code sample</summary>

```dart
import 'package:flutter_test/flutter_test.dart';
import 'package:google_fonts/google_fonts.dart';

void main() {
  TestWidgetsFlutterBinding.ensureInitialized();
  final expectedStyle = GoogleFonts.getFont('ABeeZee');

  test('test fonts', () {
    final styleFunc = GoogleFonts.asMap()['ABeeZee']!;
    expect(styleFunc(), equals(expectedStyle));
  });
}
```

</details>

<details>
  <summary>Logs</summary>

```
N/A - Error happens in tests
```

```
No issues found! (ran in 1.6s)
```

```
[✓] Flutter (Channel stable, 3.0.1, on macOS 12.3.1 21E258 darwin-arm, locale en-AU)
    • Flutter version 3.0.1 at /<path_to_flutter>/3.0.1
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision fb57da5f94 (31 hours ago), 2022-05-19 15:50:29 -0700
    • Engine revision caaafc5604
    • Dart version 2.17.1
    • DevTools version 2.12.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.0.0)
    • Android SDK at /<path_to_android>/Android/sdk
    • Platform android-32, build-tools 32.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.4)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7772763)

[✓] VS Code (version 1.67.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.40.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-arm64   • macOS 12.3.1 21E258 darwin-arm
    • Chrome (web)    • chrome • web-javascript • Google Chrome 101.0.4951.64

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!
```

</details>

### PR DESCRIPTION
**Pull request by guidezpl**
Improve testing guidance and asset manifest loading errors 

Fixes #244

### CODE REVIEW

File: example/test/unit_test.dart [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+// Consider `flutter test --no-test-assets` if assets are not required.
+void main() {
+  TestWidgetsFlutterBinding.ensureInitialized();
+  // This has the side effect of setting up a mock HTTP client.
+  // Disable this with HttpOverrides.global = null;
+
+  test('Can test fonts', () {
+    final styleFunc = GoogleFonts.asMap()['ABeeZee']!;
+    final expectedStyle = GoogleFonts.getFont('ABeeZee');
```
Comment (zeshuaro) [inline example/test/unit_test.dart:12]: The code snippet I shared was just one test case to show the issue I was facing. I also experimented this a bit, and I wonder if it makes more sense to put this in `setUp` or `setUpAll` to account for a bigger suite of tests? I tried this out, and it seems to be working too. So for example:

```dart
import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';
import 'package:google_fonts/google_fonts.dart';

void main() {
  TestWidgetsFlutterBinding.ensureInitialized();
  late TextStyle expectedStyle;

  setUpAll(() {
    expectedStyle = GoogleFonts.getFont('ABeeZee');
  });

  test('test fonts', () {
    final styleFunc = GoogleFonts.asMap()['ABeeZee']!;
    expect(styleFunc(), equals(expectedStyle));
  });
}
```

What do you think?

File: lib/src/google_fonts_base.dart [hunk 2]
```diff
@@ -175,12 +175,20 @@ Future<void> loadFontIfNecessary(GoogleFontsDescriptor descriptor) async {
...
-      print(
-        'See https://docs.flutter.dev/development/data-and-backend/networking#platform-notes.',
-      );
+        'following exception occurred:\n$e');
+    if (file_io.isTest) {
+      print('\nThere is likely something wrong with your test. Please see '
+          'https://github.com/material-foundation/google-fonts-flutter/blob/main/example/test '
+          'for examples of how to test with google_fonts.');
+    } else {
+      if (file_io.isMacOS || file_io.isAndroid) {
```
Comment (zeshuaro) [inline lib/src/google_fonts_base.dart:184]: /nit this can be an `else if`?

**Review by zeshuaro — APPROVED** [review]
Thanks for the quick update to address my issue, just left some minor comments, otherwise LGTM.

File: example/test/unit_test.dart (hunk 1 above)
Comment (guidezpl) [reply example/test/unit_test.dart:12]: I prefer showing this slightly more complex example to avoid people encoutering the same issue you did :thumbs_up

**Pull request merged**